File: src/main/java/org/jukeboxmc/network/handler/ContainerCloseHandler.java
package org.jukeboxmc.network.handler;

import com.nukkitx.protocol.bedrock.packet.ContainerClosePacket;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.Server;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ContainerCloseHandler implements PacketHandler<ContainerClosePacket>{

    @Override
    public void handle(@NotNull ContainerClosePacket packet, Server server, @NotNull Player player ) {
        player.setClosingWindowId( packet.getId() );
        player.closeInventory( packet.getId(), packet.isUnknownBool0() );
        player.setClosingWindowId( Integer.MIN_VALUE );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/ContainerCloseHandler.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.packet.ContainerClosePacket
import org.jukeboxmc.server.JukeboxServer
import org.jukeboxmc.server.player.JukeboxPlayer

class ContainerCloseHandler : PacketHandler<ContainerClosePacket> {

    override fun handle(packet: ContainerClosePacket, server: JukeboxServer, player: JukeboxPlayer) {
        player.closeInventory(packet.id, packet.isServerInitiated)
    }
}